### ISSUE TITLE
av-store: granular pruning of data

### ISSUE THREAD

**Issue by sandreim** [paritytech/polkadot#7237]
Currently we block the subsystem for whatever times it takes to prune the data both on timer and on finality. We should make this process more granular as to allow the processing of messages in between otherwise we can end up in a situation where this takes > 10s and node crashes due to `SubsystemStalled` error.

https://github.com/paritytech/polkadot/blob/19fdd197aff085f7f66e54942999fd536e7df475/node/core/av-store/src/lib.rs#L617

In the past I have seen very high times up to 10s on a few nodes, but not as of recently on tests with small PoVs.

<img width="1494" alt="Screenshot 2023-05-16 at 17 17 35" src="https://github.com/paritytech/polkadot/assets/54316454/202726e2-a22f-4096-8e08-edfd9e8026df">

**Comment by bkchr** [comment]
Do you know what takes the time? The actual IO when applying the TX? Or the processing of what to delete?  If it is the later, we should just move this to some background process.

**Comment by sandreim** [comment]
> Do you know what takes the time? The actual IO when applying the TX? Or the processing of what to delete? If it is the later, we should just move this to some background process.

I am not sure which part exactly is heavy. Before moving this as a background task we'd have to be certain that we don't corrupt up the db with multiple writers/readers on same column.

**Comment by bkchr** [comment]
If the db is Sync & Send, I would assume that you can read and write from multiple threads :P

**Comment by sandreim** [comment]
Yes it is, but usually we do this from a single subsystem thread. Maybe in this case, it shouldn't be an issue as the keys should no longer be accessed by anything since this will happen after 25h hours.

**Comment by alexggh** [comment]
Moving prune_all to a separate blocking task: https://github.com/paritytech/polkadot/pull/7263

**Comment by alexggh** [comment]
Fixed with: https://github.com/paritytech/polkadot/pull/7263
